### DISCUSSION TITLE
That whole notion of time grain and aggregate functions for time series vs. raw data

### DISCUSSION THREAD

**Discussion by arminus** [apache/superset#18964] [Q&A / Help]
So I have two gripes with the whole approach of time grain and apparently enforced aggregate functions for time series:

- first of all, it took me an hour or so to realize that charts get completely wrong if the 10K limit is reached, see below (I'm using an AVG on the value for this metric and a postgres database table)
- secondly, what is the rational behind [apparently enforcing](https://github.com/apache/superset/discussions/18755) the use of aggregate functions?

While I understand the need to average values with time grain != original timestamp, why is there apparently no way to avoid that when keeping the original timestamps (and staying below the 10K "limit")? in my case, this always averages away a couple of extremes/outliers I have in my data which is kind of bad for the visualization I'm after

But maybe I'm missing something completely here...?

![Superset-Timeseries](https://user-images.githubusercontent.com/683680/156002542-d1a23491-4057-4e3b-86e7-12f2cda058cd.gif)

**Comment by sfirke** [comment]
Re: your first bullet, yes this is bad and has bitten me before.  I believe this PR https://github.com/apache/superset/pull/27700 recently made it more obvious when the chart may be inaccurate due to the row limit being reached, though I haven't tested it.

On your second point, there's not really a rationale for it, I don't think at this point anyone is objecting to adding the functionality to see raw data without aggregation.  But it would be a ton of work given the assumptions baked into Superset, and I don't think it's a priority for any developers who are capable of pulling it off.  More context in the thread and final comment here: https://github.com/apache/superset/discussions/19182
